CMakeLists.txt

```diff
@@ -0,0 +1,122 @@
+message(STATUS "Using LLVMConfig.cmake in: ${LLVM_DIR}")
+
+include(${LLVM_DIR}/LLVMConfig.cmake)
+
+list(APPEND CMAKE_MODULE_PATH ${LLVM_DIR})
+include(AddLLVM)
+
+include(SwiftSupport)
+include(GNUInstallDirs)
+
+set(swift_optimization_flags)
+if(CMAKE_BUILD_TYPE MATCHES Release)
+  set(swift_optimization_flags -O)
+endif()
+
+add_swift_library(XCTest
+                  MODULE_NAME
+                    XCTest
+                  MODULE_LINK_NAME
+                    XCTest
+                  MODULE_PATH
+                    ${CMAKE_CURRENT_BINARY_DIR}/swift/XCTest.swiftmodule
+                  OUTPUT
+                    ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_SHARED_LIBRARY_PREFIX}XCTest${CMAKE_SHARED_LIBRARY_SUFFIX}
+                  LINK_FLAGS
+                    -L${XCTEST_PATH_TO_LIBDISPATCH_BUILD}/src -ldispatch
+                    -L${XCTEST_PATH_TO_FOUNDATION_BUILD} -lFoundation
+                  SOURCES
+                    Sources/XCTest/Private/WallClockTimeMetric.swift
+                    Sources/XCTest/Private/TestListing.swift
+                    Sources/XCTest/Private/XCTestCaseSuite.swift
+                    Sources/XCTest/Private/TestFiltering.swift
+                    Sources/XCTest/Private/XCTestInternalObservation.swift
+                    Sources/XCTest/Private/ObjectWrapper.swift
+                    Sources/XCTest/Private/PerformanceMeter.swift
+                    Sources/XCTest/Private/PrintObserver.swift
+                    Sources/XCTest/Private/ArgumentParser.swift
+                    Sources/XCTest/Private/XCPredicateExpectation.swift
+                    Sources/XCTest/Public/XCTestRun.swift
+                    Sources/XCTest/Public/XCTestMain.swift
+                    Sources/XCTest/Public/XCTestCase.swift
+                    Sources/XCTest/Public/XCTestSuite.swift
+                    Sources/XCTest/Public/XCTestSuiteRun.swift
+                    Sources/XCTest/Public/XCTestErrors.swift
+                    Sources/XCTest/Public/XCTestObservation.swift
+                    Sources/XCTest/Public/XCTestCaseRun.swift
+                    Sources/XCTest/Public/XCAbstractTest.swift
+                    Sources/XCTest/Public/XCTestObservationCenter.swift
+                    Sources/XCTest/Public/XCTestCase+Performance.swift
+                    Sources/XCTest/Public/XCTAssert.swift
+                    Sources/XCTest/Public/Asynchronous/XCTestCase+NotificationExpectation.swift
+                    Sources/XCTest/Public/Asynchronous/XCTestCase+PredicateExpectation.swift
+                    Sources/XCTest/Public/Asynchronous/XCPredicateExpectationHandler.swift
+                    Sources/XCTest/Public/Asynchronous/XCWaitCompletionHandler.swift
+                    Sources/XCTest/Public/Asynchronous/XCNotificationExpectationHandler.swift
+                    Sources/XCTest/Public/Asynchronous/XCTestCase+Asynchronous.swift
+                    Sources/XCTest/Public/Asynchronous/XCTestExpectation.swift
```

Comment:
(I vaguely remembered that globbing source files in CMake was discouraged, so for any other reviewers out there, I  looked up the exact reason: https://stackoverflow.com/a/32412044/679254)